pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -118,10 +113,8 @@ func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{}, client.IgnoreNotFound(err)
 	}
 
-	if !workload.HasQuotaReservation(wl) || apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadFinished) {
-		// 1.2 workload has no reservation or is finished
-		log.V(5).Info("workload with no reservation, delete owned requests")
-		return reconcile.Result{}, c.deleteOwnedProvisionRequests(ctx, req.Namespace, req.Name)
+	if !workload.HasQuotaReservation(wl) || workload.IsFinished(wl) {
```

Comment:
I suppose this is not necessary for the CapacityRevoked support. If so, can we extract to a dedicated PR?

Reply:
It actually is. If MaxRunDuration time will be exceeded, workload may get finished before ClusterAutoscaler sets CapacityRevoked. Hence we may not be able to sync with the conditions, set appropriate status and emit an event

Reply:
Ah, got it, so we keep ProvReq to give CA time to set the condition. I think the relationship is not obvious, please explain the motivation for coupling the changes in the PR description.